tests/sentry/ratelimits/test_redis_concurrent.py

```diff
@@ -34,13 +39,16 @@ def test_add_and_remove(self):
             self.backend.finish_request("foo", "request_id1")
             assert self.backend.get_concurrent_requests("foo") == limit - 1
 
-    def test_fails_open(self):
+    def test_fails_open(self) -> None:
         class FakeClient:
-            def __init__(self, real_client):
+            def __init__(
+                self,
+                real_client: Any,  # RedisCluster[str] | StrictRedis[str]
+            ) -> None:
                 self._client = real_client
 
-            def __getattr__(self, name):
-                def fail(*args, **kwargs):
+            def __getattr__(self, name: str) -> Any:
+                def fail(*args: Any, **kwargs: Any) -> None:
                     raise Exception("OH NO")
 
                 return fail
```

Comment:
perhaps `-> Callable[..., Never]` and `-> Never` for the inner function

though really this could be replaced with `mock.Mock(side_effect=...)` probably